src/std/net/s3/interface.ss

```diff
@@ -0,0 +1,24 @@
+(import :std/interface
+        :std/contract
+        :std/misc/alist)
+
+(export #t)
+
+(interface S3
+  (get-bucket (name :~ string?))
+  (create-bucket! (name :~ string?)
+         (opts :~ (maybe alist?) := #f))
+  (delete-bucket! (name :~ string?))
+  (bucket-exists? (name :~ string?))
+  (list-buckets))
+
+(interface S3Bucket
+  (get  (name :~ string?))
+  (put! (name :~ string?)
+        (data :~ u8vector?)
+        content-type: (content-type := "octet-stream" :~ string?))
```

Comment:
do we care about this?

Note that dynamic dispatch with kw lambdas _is_ expensive.

Reply:
Do we pay that cost no matter what or only when we specify `content-type:`?

Reply:
I guess it's not really necessary if we're enforcing `(data :~ u8vector?)`

Reply:
Yes, it is a dynamic call through the interface, so the compiler cannot optimize dispatch.